src/draw/vg_lite/lv_draw_vg_lite_layer.c

```diff
@@ -45,15 +45,15 @@ void lv_draw_vg_lite_layer(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t
 
     /*It can happen that nothing was draw on a layer and therefore its buffer is not allocated.
      *In this case just return. */
-    if(layer->buf == NULL)
+    if(layer->draw_buf == NULL)
         return;
 
     lv_image_dsc_t img_dsc;
     lv_memzero(&img_dsc, sizeof(lv_image_dsc_t));
     img_dsc.header.w = lv_area_get_width(&layer->buf_area);
     img_dsc.header.h = lv_area_get_height(&layer->buf_area);
-    img_dsc.header.cf = layer->color_format;
-    img_dsc.data = layer->buf;
+    img_dsc.header.cf = layer->draw_buf->header.cf;
```

Comment:
Remove img_dsc, call `lv_draw_vg_lite_img(draw_unit, layer->draw_buf, coords); ` directly.

Please also check if `LV_IMAGE_FLAGS_PREMULTIPLIED ` is already set correctly.